spec/controllers/groups_controller_spec.rb

```diff
@@ -287,13 +287,114 @@
         end
       end
     end
-    describe '#invalidate_groupings'
-    describe '#validate_groupings'
-    describe '#delete_groupings'
-    describe '#add_members'
-    describe '#add_member'
-    describe '#remove_members'
-    describe '#remove_member'
+
+    describe '#validate_groupings' do
+      let(:grouping) { create :grouping_with_inviter }
+
+      it 'should validate groupings' do
+        post :global_actions, params: {
+            assignment_id: grouping.assignment.id,
+            groupings: [grouping.id],
+            global_actions: 'valid'
+        }
+        expect(grouping.reload.admin_approved).to be true
+      end
+    end
+
+    describe '#invalidate_groupings' do
+      let(:grouping) { create :grouping_with_inviter }
+
+      it 'should invalidate groupings' do
+        post :global_actions, params: {
+            assignment_id: grouping.assignment.id,
+            groupings: [grouping.id],
+            global_actions: 'valid'
+        }
+        expect(grouping.reload.admin_approved).to be true
+
+        post :global_actions, params: {
+            assignment_id: grouping.assignment.id,
+            groupings: [grouping.id],
+            global_actions: 'invalid'
+        }
+
+        expect(grouping.reload.admin_approved).to be false
+      end
+    end
+
+    describe '#delete_groupings' do
+      before :each do
+        @grouping = create :grouping_with_inviter
```

Comment:
please use `let` or `let!` instead of setting instance attributes for tests whenever possible